go/rpcclientlib/viewing_key_client.go

```diff
@@ -34,34 +29,43 @@ func NewViewingKeyClient(client Client) (*ViewingKeyClient, error) {
 	enclavePublicKey := ecies.ImportECDSAPublic(enclPubECDSA)
 
 	vkClient := &ViewingKeyClient{
-		obscuroClient:      client,
-		enclavePublicKey:   enclavePublicKey,
-		viewingKeysPrivate: map[common.Address]*ecies.PrivateKey{},
-		viewingKeysPublic:  map[common.Address][]byte{},
+		obscuroClient:     client,
+		enclavePublicKey:  enclavePublicKey,
+		viewingKeyManager: vkManager,
 	}
 
 	return vkClient, nil
 }
 
-// NewViewingKeyNetworkClient returns a network RPC client with Viewing Key encryption/decryption
-func NewViewingKeyNetworkClient(rpcAddress string) (*ViewingKeyClient, error) {
+// NewViewingKeyNetworkClient returns a network RPC client with Viewing Key encryption/decryption for a single account
+func NewViewingKeyNetworkClient(rpcAddress string, wallet wallet.Wallet, viewingKey *ecies.PrivateKey, vkPublic []byte, signedVK []byte) (*ViewingKeyClient, error) {
```

Comment:
this would be: `NewAuthClient(rpcAddress, viewingKey ViewingKey)`
- basically to use the viewingkey type which contains everything, including the address